native-shared-library/README

```diff
@@ -0,0 +1,81 @@
+# Build a Shared Library with Native Image
+
+This demo shows how to create a small Java class library, use `native-image` to create a native shared library, and then create a small C application that uses that shared library.
```

Comment:
@LesiaChaban, please move all content from the website guide https://www.graalvm.org/dev/reference-manual/native-image/guides/build-native-shared-library/. Add missing pieces in the introduction part.